Copter, Rover, Tracker, Plane-4.5.7 release

rmackay9 opened the pull request:
This is the Copter, Rover, Tracker, Plane-4.5.7 release.  The contents of that release can be found in the [4.5.7-beta1 column of this project](https://github.com/orgs/ArduPilot/projects/8/views/8?layout=board).

This is the same as the 4.5.7-beta1 release but with PR https://github.com/ArduPilot/ardupilot/pull/27799 reverted.  I have not actually determined if [the user report](https://discuss.ardupilot.org/t/coper-4-5-7-beta1-available-for-testing/124355) is correct but I think at this stage it is best to remove it and proceed with the release.

SeptentrioGNSS commented:
@rmackay9 could you explain why the DNU value of SVnum commit into AP_GPS was reverted?

SeptentrioGNSS commented:
Hi @tridge, I am still waiting on an explanation on why this has been reverted back from #27799, could you help on this please? 

The case this was trying to solve was when (under specific circumstance) the reported value turns from a numeric (e.g 25) to the 255 value from one message to the next. With the != approach, the previous value will stay in play and users will not have the info that the Septentrio GNSS receiver is not tracking any satellites. With the approach from @flyingthingsintothings and @chiara-septentrio they solved this as the NrSV value will be fixed to 0. 

With the users report on the forum (mentioned by @rmackay9 in the description here) the discussion goes towards the [HDOP](https://gssc.esa.int/navipedia/index.php?title=Positioning_Error#:~:text=Horizontal%20Dilution%20of%20Precision) value (which is inherently different from NrSV and comes from a completely different [SBF message](https://www.septentrio.com/en/support/product-resources)) it is set to 'zero' when it is not available. Otherwise you have to divide by 100 to get the actual value (also this division I do not see in the code upon parsing the message). 

To give an example, here are consecutive PVTGeodetic blocks with the NrSV values dropping to 255 while the antenna is disconnected (rare use case but still plausible and should be indicated to user other than keeping the latest 'normal value') from the Septentrio GNSS receiver.

![nrsv_septentrio](https://github.com/user-attachments/assets/ab2c58e0-acf7-439c-97bf-c85f1a7ef111)

rmackay9 commented:
Hi @SeptentrioGNSS,

We reverted the change because we couldn't determine quickly whether the issue was a real problem or not.  We didn't want to hold up other fixes so we reverted the one questionable change and proceeded with the release.

The change is included in [4.6 which is undergoing beta testing now](https://discuss.ardupilot.org/t/copter-4-6-0-beta2-is-available-for-beta-testing/127445).  It's hard to exactly predict when 4.6 will be released as the stable version but I think within a month is possible.

We could try to include the change again if we do another 4.5.x release as well.  There's not definitive plans to do that at the moment though.

As AP has become more stable and mature, we struggle to find enough beta testers.  Any and all feedback during beta testing is greatly appreciated!
